runner/internal/shim/api/http.go

```diff
@@ -51,3 +51,20 @@ func (s *ShimServer) PullGetHandler(w http.ResponseWriter, r *http.Request) (int
 		State: string(s.runner.GetState()),
 	}, nil
 }
+
+func (s *ShimServer) StopPostHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
```

Comment:
Why don't we check the state?

Reply:
The state checks in the `DockerRunner.Stop`. I've added the check. But we need to rework this code.